pippy/PipelineSchedule.py

```diff
@@ -74,22 +74,225 @@ def compute_loss(self):
 
 
 def create_buffers(
-    input: Union[torch.Tensor, List[torch.tensor]], device: torch.device
+    tensor: Union[torch.Tensor, List[torch.tensor]], device: torch.device
 ) -> List[torch.Tensor]:
     """
-    Creates buffers for a given input on a specified device.
+    Creates buffers for a given tensor on a specified device.
     This function takes as input a tensor or a list of tensors and returns a tensor or a list of tensors (respectively)
     of the same shape, but located on the specified device and uninitialized (i.e., filled with arbitrary data).
     """
-    if isinstance(input, torch.Tensor):
-        return [torch.empty_like(input, device=device)]
-    elif isinstance(input, (list, tuple)):
-        return [torch.empty_like(inp, device=device) for inp in input]
-    raise ValueError(
-        f"Unsupported input type {type(input)} cannot create buffers"
+    if isinstance(tensor, torch.Tensor):
+        return [torch.empty_like(tensor, device=device)]
+    elif isinstance(tensor, (list, tuple)):
+        return [torch.empty_like(t, device=device) for t in tensor]
+    raise TypeError(
+        f"Unsupported input type {type(tensor)} cannot create buffers"
     )
 
 
+METADATA_TENSOR_LEN = 50
+PLACEHOLDER_VAL = 0
```

Comment:
nit: shall we reserve value 0 for scalar tensor? (Just in case)
```
>>> a = torch.tensor(1)
>>> a.shape
torch.Size([])
>>> len(a.shape)
0
```

Reply:
This is a good edge case, i hadnt considered it.

I changed the placeholder value to -1 and also added a new test which considers this shape as well. Thanks!